test/functional/fixtures/api/es-next/user-variables/test.js

```diff
@@ -0,0 +1,10 @@
+const DEFAULT_SELECTOR_TIMEOUT   = 3000;
+const DEFAULT_RUN_OPTIONS        = {
+    selectorTimeout: DEFAULT_SELECTOR_TIMEOUT,
+};
+
+describe('[API] UserVariables', function () {
+    it('Should provide user variables', function () {
+        return runTests('./testcafe-fixtures/user-variables-test.js', 'test', DEFAULT_RUN_OPTIONS);
+    });
+});
```

Comment:
You added the test covering the 'read; capability. Add the test for 'write' capability.
Like this one:

```js
import { userVariables } from 'testcafe';

fixture `Write access`
    .before({
        userVariables.executedTests = [];
     })
    .after(() => {
      await t.expect(userVariables.executedTests).eql(['1', '2', '3']);
   })

test('1', async t => {
   userVariables.executedTests.push('1');
});

test('2', async t => {
      userVariables.executedTests.push('2');
});

test('3', async t => {
        userVariables.executedTests.push('3');
});
```